Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -165,6 +162,15 @@ public actor SwiftLanguageService: LanguageService, Sendable {
 
   private var diagnosticReportManager: DiagnosticReportManager!
 
+  var documentManager: DocumentManager {
+    get {
+      guard let sourceKitLSPServer = self.sourceKitLSPServer else {
+        fatalError("Cannot get document manager because sourceKitLSPServer was deallocated")
+      }
```

Comment:
We should not crash the sourcekit-lsp server process if `sourceKitLSPServer` got deallocated. It’s a weak variable and it becoming `nil` is a valid thing that can happen. Instead, we should throw an error if it becomes `nil`, which will return an error response for the requests. You should be able to make this getter throwing by annotating it as `get throws`.

Reply:
If it throws we'd still have to wrap a `try` around documentManger everytime we access that right? arguably the same workload with wrapping a `guard let` when marking it as optional like we discussed earlier?

Reply:
Yes, exactly. I think adding `try` is a lot more lightweight than `guard let …. else { throw ResponseError.unknown("some message") }`. 

Many methods in `SwiftLanguageService` throw already anyway (eg. all the calls to `documentManager.latestSnapshot`) and for those methods, it’s just a matter of adding the `try` keyword if it doesn’t exist yet. For other methods, it should be fairly simple to log the failure by wrapping the access to `documentManager` in a `orLog` which will log the error and return `nil` if getting `documentManager` throws.